modules/Sources/Dependencies/AddressBookClient/AddressBookLiveKey.swift

```diff
@@ -54,39 +62,46 @@ extension AddressBookClient: DependencyKey {
                     let encryptedFileURL = documentsDirectory.appendingPathComponent(try AddressBookClient.filenameForEncryptedFile())
 
                     if let contactsData = try? Data(contentsOf: encryptedFileURL) {
-                        // file exists, try to find the unencrypted file in case the delete failed
+                        let contacts = try AddressBookClient.contactsFrom(encryptedData: contactsData)
+                        
+                        // file exists and was successfully decrypted and parsed;
+                        // try to find the unencrypted file and delete it
                         let unencryptedFileURL = documentsDirectory.appendingPathComponent(Constants.unencryptedFilename)
                         if FileManager.default.fileExists(atPath: unencryptedFileURL.path) {
                             try? FileManager.default.removeItem(at: unencryptedFileURL)
                         }
-                        
-                        let contacts = try AddressBookClient.contactsFrom(encryptedData: contactsData)
 
                         latestKnownContacts = contacts
-                        return contacts
+                        return (contacts, nil)
                     } else {
                         // Fallback to the unencrypted file check and resolution
                         let unencryptedFileURL = documentsDirectory.appendingPathComponent(Constants.unencryptedFilename)
                         
                         if let contactsData = try? Data(contentsOf: unencryptedFileURL) {
-                            // file exists, ensure data are parsed, re-saved and file deteled
-                            let contacts = try AddressBookClient.contactsFrom(unencryptedData: contactsData)
+                            // Unencrypted file exists; ensure data are parsed, re-saved as encrypted, and the original file deleted.
+
+                            var contacts = try AddressBookClient.contactsFrom(contactsData)
```

Comment:
When I suggested changing the `unencryptedData:` label in `AddressBookClient.contactsFrom(unencryptedData:)`, I had not seen that there was also a `contactsFrom(encryptedData:)` method. The two methods are now more confusable. Please either revert that change or rename `contactsFrom(_)` to something like `AddressBookClient.decodeContacts`. Sorry for going back and forth on this.

Reply:
Updated